states/kv/kv.go

```diff
@@ -0,0 +1,595 @@
+	key = path.Join(kv.rootPath, key)
+
+	ss := kv.client.GetSnapshot(MaxSnapshotTS)
+	ss.SetScanBatchSize(SnapshotScanSize)
+
+	val, err := ss.Get(ctx, []byte(key))
+	if err != nil {
+		return "", errors.Wrap(err, fmt.Sprintf("Failed to load value for key %s", key))
+	}
+
+	// Check if value is the empty placeholder
+	value := convertEmptyByteToString(val)
+
+	return value, err
+}
+
+// LoadWithPrefix returns all the keys and values for the given key prefix.
+func (kv *txnTiKV) LoadWithPrefix(ctx context.Context, prefix string) ([]string, []string, error) {
+	prefix = path.Join(kv.rootPath, prefix)
+
+	ss := kv.client.GetSnapshot(MaxSnapshotTS)
+	ss.SetScanBatchSize(SnapshotScanSize)
+
+	// Retrieve key-value pairs with the specified prefix
+	startKey := []byte(prefix)
+	endKey := tikv.PrefixNextKey([]byte(prefix))
+	iter, err := ss.Iter(startKey, endKey)
+	if err != nil {
+		err = errors.Wrap(err, fmt.Sprintf("Failed to create iterater for LoadWithPrefix() for prefix: %s", prefix))
+		return nil, nil, err
+	}
+	defer iter.Close()
+
+	var keys []string
+	var values []string
+
+	// Iterate over the key-value pairs
+	for iter.Valid() {
+		val := iter.Value()
+		// Check if empty value placeholder
+		strVal := convertEmptyByteToString(val)
+		keys = append(keys, string(iter.Key()))
+		values = append(values, strVal)
+		err = iter.Next()
+		if err != nil {
+			err = errors.Wrap(err, fmt.Sprintf("Failed to iterate for LoadWithPrefix() for prefix: %s", prefix))
+			return nil, nil, err
+		}
+	}
+	return keys, values, nil
+}
+
+// Save saves the input key-value pair.
+func (kv *txnTiKV) Save(ctx context.Context, key, value string) error {
+	key = path.Join(kv.rootPath, key)
+
+	txn, err := kv.client.Begin()
+	if err != nil {
+		return errors.Wrap(err, "Failed to build transaction for putTiKVMeta")
+	}
+
+	// Check if the value being written needs to be empty plaeholder
+	byteValue, err := convertEmptyStringToByte(value)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to cast to byte (%s:%s) for putTiKVMeta", key, value))
+	}
+	err = txn.Set([]byte(key), byteValue)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to set value for key %s in putTiKVMeta", key))
+	}
+	return txn.Commit(ctx)
+}
+
+// Remove removes the input key.
+func (kv *txnTiKV) Remove(ctx context.Context, key string) error {
+	key = path.Join(kv.rootPath, key)
+
+	txn, err := kv.client.Begin()
+	if err != nil {
+		return errors.Wrap(err, "Failed to build transaction for removeTiKVMeta")
+	}
+
+	err = txn.Delete([]byte(key))
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to remove key %s in removeTiKVMeta", key))
+	}
+	return txn.Commit(ctx)
+}
+
+// RemoveWithPrefix removes the keys for the given prefix.
+func (kv *txnTiKV) RemoveWithPrefix(ctx context.Context, prefix string) error {
+	prefix = path.Join(kv.rootPath, prefix)
+	ctx, cancel := context.WithTimeout(ctx, RequestTimeout)
```

Comment:
NIT: `RequestTimeout` shall be deprecated since ctx could be controlled from command